isotp/esp_isotp/esp_isotp.h

```diff
@@ -0,0 +1,105 @@
+#ifndef _ISOTP_ESP_H_
+#define _ISOTP_ESP_H_
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/queue.h"
+#include "esp_err.h"
+#include "esp_twai.h"
+#include "esp_twai_types.h"
+#include "isotp-c/isotp.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief ISO-TP link structure
+ */
+struct esp_isotp_link {
+    IsoTpLink link;
+    uint8_t *tx_buffer;
+    uint8_t *rx_buffer;
+    uint16_t tx_buffer_size;
+    uint16_t rx_buffer_size;
+    QueueHandle_t rx_queue;
+    twai_node_handle_t twai_node;
+};
+typedef struct esp_isotp_link esp_isotp_link_t;
+typedef struct esp_isotp_link *esp_isotp_handle_t;
+
+/**
+ * @brief Structure to hold received ISO-TP queue item
+ */
+struct isotp_rx_queue_item {
+    twai_frame_header_t header;
+    uint8_t data[TWAI_FRAME_MAX_LEN];
+    uint8_t data_len;
+};
+typedef struct isotp_rx_queue_item isotp_rx_queue_item_t;
+
+/**
+ * @brief Configuration structure for creating a new ISO-TP link
+ */
+struct esp_isotp_config {
+    uint32_t tx_id;                  /*!< ID for transmitting messages */
+    uint32_t rx_id;                  /*!< ID for receiving messages */
+    uint16_t tx_buffer_size;
+    uint16_t rx_buffer_size;
+    twai_node_handle_t twai_node;    /*!< TWAI node handle */
+    QueueHandle_t rx_queue;          /*!< Queue for receiving messages */
+};
+typedef struct esp_isotp_config esp_isotp_config_t;
+
+/**
+ * @brief Create a new ISO-TP link
+ *
+ * @param config Pointer to the configuration structure
+ * @return Pointer to the new ISO-TP link, or NULL on error
+ */
+esp_isotp_handle_t esp_isotp_new(const esp_isotp_config_t *config);
```

Comment:
```suggestion
esp_isotp_handle_t esp_isotp_new_link(twai_node_handle_t twai_node, const esp_isotp_config_t *config);
```

or


```suggestion
esp_isotp_handle_t esp_isotp_new_transport(twai_node_handle_t twai_node, const esp_isotp_config_t *config);
```

Reply:
becasue ISO-TP is transport layer protocol, right? maybe use "new_transport" is more clear 🤔